runtime/lua/vim/lsp/util.lua

```diff
@@ -800,13 +800,15 @@ function M.fancy_floating_markdown(contents, opts)
   local width, height = M._make_floating_popup_size(stripped, opts)
 
   -- Insert blank line separator after code block
-  local insert_separator = opts.separator or true
+  local insert_separator = opts.separator
```

Comment:
doesn't that always enable the separator ? maybe we should set insert_separator to the char to insert (eg "─" by default) and if null don't print a line.

Reply:
that's exactly what the next line is for :)

Reply:
I thing I get it, it should be disabled only if opts.separator is false ? (was thinking setting it to nil was a valid way to disable as well which confused me).

Reply:
Exactly, but `nil` means that the option isn't set, and the default (true) should be used. But the disjunction of course always returned `true` (since nil is false-y).